lightkurve/correctors/cbvcorrector.py

```diff
@@ -1,258 +1,946 @@
+                interpolated to the light curve cadence times
+        """
+
+        if not isinstance(lc, LightCurve):
+            raise Exception('<lc> must be a LightCurve class')
+        
+        # Creat the new cbv object with no basis vectors, yet...
+        cbvNewTime = lc.time.copy()
+        # Gaps are all false
+        gaps = np.full(len(lc.time), False)
+        dataTbl = Table([lc.cadenceno, gaps], names=('CADENCENO', 'GAP'))
+
+        # We are PCHIP interpoalting each CBV independently.
+        for idx in self.cbv_indices:
+            fInterp = PchipInterpolator(self.time.value,
+                    self['VECTOR_{}'.format(idx)], extrapolate=extrapolate)
+            dataTbl['VECTOR_{}'.format(idx)] = fInterp(lc.time.value)
+
+        dataTbl.meta = self.meta.copy()
+
+        # We need to return a new CotrendingBasisVectors class. Make sure we
+        # instatiate the correct type.
+        if isinstance(self, KeplerCotrendingBasisVectors):
+            return KeplerCotrendingBasisVectors(data=dataTbl, time=cbvNewTime)
+        elif isinstance(self, TessCotrendingBasisVectors):
+            return TessCotrendingBasisVectors(data=dataTbl, time=cbvNewTime)
+        else:
+            return CotrendingBasisVectors(data=dataTbl, time=cbvNewTime)
+
+#***
+class KeplerCotrendingBasisVectors(CotrendingBasisVectors):
+    """ Sub-class for Kepler/K2 cotrending basis vectors
+
+    See CotrendingBasisVectors for class details
+
+    Attributes
+    ----------
+    CotrendingBasisVectors attributes
+    astropy.timeseries.TimeSeries attributes
+    mission         : [str] ('Kepler', 'K2')
+    cbv_type        : [str] always 'SingleScale'
+    quarter         : [int] Kepler Quarter
+    campaign        : [int] K2 Campaign
+    module          : [int] Kepler instrument CCD module
+    output          : [int] Kepler instrument CCD output
 
-        References
+    """
+
+    #***
+    validMissionOptions = ('Kepler', 'K2')
+    validCBVTypes = ('SingleScale')
+
+    #***
+
+    def __init__(self, data=None, time=None, **kwargs):
+        """ Initiatates a KeplerCotrendingBasisVectors object.
+        Normally one would use KeplerCotrendingBasisVectors.from_HDU to
+        automatically set up the object. However, for certain functionality
+        one must instantiate the object directly.
+        """
+
+        # Normally these extra meta fields will be set up in the from_HDU
+        # classmethod, if they are not, then set to default values so that the
+        # @property methods do not error
+        if data is not None:
+            if not data.meta.__contains__('MISSION'):
+                data.meta['MISSION'] = None
+            if not data.meta.__contains__('CBV_TYPE'):
+                data.meta['CBV_TYPE'] = None
+            if not data.meta.__contains__('MODULE'):
+                data.meta['MODULE'] = None
+            if not data.meta.__contains__('OUTPUT'):
+                data.meta['OUTPUT'] = None
```

Comment:
I recommend removing lines 393-404 and instead change the property methods such that they return `self.meta.get(key, default=None)` instead of `self.meta[key]`.  The former returns `None` and the latter raises an exception if `key` is missing.  It means we can have less code :smiley:

Reply:
Removed.